### PR TITLE
ARTEMIS-1463 - add role based authentication to the JMX objects

### PR DESCRIPTION
**Pull request by andytaylor**
This is done by creating a guard and using JAAS to check for access to mbean objects and their methods.

NB this also implements https://issues.apache.org/jira/browse/ARTEMIS-534

https://issues.apache.org/jira/browse/ARTEMIS-1463

### CODE REVIEW

File: artemis-cli/pom.xml [hunk 1]
```diff
@@ -35,6 +35,11 @@
    <dependencies>
       <dependency>
          <groupId>org.apache.activemq</groupId>
+         <artifactId>artemis-boot</artifactId>
```
Comment (clebertsuconic) [inline artemis-cli/pom.xml]: why do you need the boot here? it should be the opposite.. this would be a circular dependency not so useful.

Comment (andytaylor) [reply artemis-cli/pom.xml]: I will remove that, it doesn't need to be there

Comment (andytaylor) [reply artemis-cli/pom.xml]: removed

**Comment by clebertsuconic** [comment]
on the dto package, I would create a sub package for this new stuff... say... security...

I would move what's in there (later or now) to a package apps... or some other name.


just to make it concise?

WDYT?

**Comment by andytaylor** [comment]
jaxb expects it to be in the same package or it doesnt compile then. I can restructure this later

**Comment by clebertsuconic** [comment]
ok.. will merge it then

File: artemis-server/pom.xml [hunk 2]
```diff
@@ -53,6 +53,11 @@
       </dependency>
       <dependency>
          <groupId>org.apache.activemq</groupId>
+         <artifactId>artemis-boot</artifactId>
```
Comment (clebertsuconic) [inline artemis-server/pom.xml]: remove this?


the boot is just a classloader thing.. no component should ever need to look into this.

Comment (andytaylor) [reply artemis-server/pom.xml]: removed

**Pull request merged**